### PR TITLE
Docker: task2-nginx

### PR DESCRIPTION
**Pull request by musfiksifat**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #311 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Added docker-compose & docker-compose-prod file and added two variants of docker file

## What was done
- Added docker-compose.yml for normal app deployment
- Added Dockerfile.prod & docker-compose.prod.yml for nginx deployment

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

**Review by musfique-m — CHANGES_REQUESTED** [review]
Please move the docker-compose*yml files to the parent folders. As later we may decide to run multiple app with the same time using docker-compose.
Also please remove the comments from the prod files.

File: react_client/todo_app/Dockerfile.prod [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+COPY package*.json .
+
+# run command to install npm
+RUN npm install
+
+# copy other files to container
+COPY . .
+
+# expose port documentation 
+EXPOSE 8080
```
[the remark this answers is not in the document]
Comment (musfiksifat) [reply react_client/todo_app/Dockerfile.prod]: moved expose command, commit: [51d18f0](https://github.com/CraftsmenLtd/coe-batch-q322/pull/314/commits/51d18f0d2c691d224cd47a8eaf51052849f4dba2)

**Comment by musfiksifat** [comment]
> Please move the docker-compose*yml files to the parent folders. As later we may decide to run multiple app with the same time using docker-compose. Also please remove the comments from the prod files.

resolved by commit [51d18f0](https://github.com/CraftsmenLtd/coe-batch-q322/pull/314/commits/51d18f0d2c691d224cd47a8eaf51052849f4dba2)

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**